packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -841,6 +841,10 @@ abstract class FlutterCommand extends Command<void> {
           ...?await usageValues,
           CustomDimensions.commandHasTerminal:
             globals.stdio.hasTerminal ? 'true' : 'false',
+          if (argParser.options.containsKey(FlutterOptions.kEnableExperiment))
+            CustomDimensions.commandUsingNullSafety: stringsArg(FlutterOptions.kEnableExperiment)
+              .contains('non-nullable')
+              .toString()
```

Comment:
nit: This is getting hard to read.

Reply:
The additional usage parameters should possibly be `Map<CustomDimentions, dynamic>` with a `toString()` before sending over to `package:usage`.

`FlutterCommand` (or elsewhere) could also have some getters for whether experiments are enabled maybe? Other suggestions?

Reply:
Added a private List<String> enabledExperiments and simplified the check. Updated to use Object and convert to String in the same place that the enums are converted